### PR TITLE
runtime: Use static_sandbox_resource_mgmt=true for TEEs

### PR DESCRIPTION
**Pull request by fidencio**
When this option is enabled the runtime will attempt to determine the appropriate sandbox size (memory, CPU) before booting the virtual machine.

As TEEs do not support memory and CPU hotplug, this approach must be used.

Fixes: #6818

### CODE REVIEW

**Comment by fidencio** [comment]
Ideally I'd like to hear from folks who're working on different TEEs, and also I'd like to wait to merge this one **after** tests are up and running for all TEEs, so we make sure we don't cause any regression, thus I'm adding the `do-not-merge` label (but I'm looking forward for reviews).

File: src/runtime/config/configuration-qemu-sev.toml.in [hunk 1]
```diff
@@ -647,4 +647,4 @@ service_offload = @DEFSERVICEOFFLOAD@
 #
 # Keys can be remotely provisioned. The Kata agent fetches them from e.g.
 # a HTTPS URL:
-#provision=https://my-key-broker.foo/tenant/<tenant-id>
\ No newline at end of file
+#provision=https://my-key-broker.foo/tenant/<tenant-id>
```
Comment (fitzthum) [inline src/runtime/config/configuration-qemu-sev.toml.in:650]: Unrelated, but do we still want these sample options at the bottom of each config? They don't really match CCv0 or watch is planned for main and currently I don't think they do anything.

**Comment by amshinde** [comment]
Sounds like a good approach to me.

File: src/runtime/Makefile [hunk 2]
```diff
@@ -602,6 +603,7 @@ USER_VARS += DEFVALIDENTROPYSOURCES
 USER_VARS += DEFSANDBOXCGROUPONLY
 USER_VARS += DEFSTATICRESOURCEMGMT
 USER_VARS += DEFSTATICRESOURCEMGMT_FC
+USER_VARS += DEFSTATICRESOURCEMGMT_TEE
```
[the remark this answers is not in the document]
Comment (fidencio) [reply src/runtime/Makefile:606]: @fitzthum, I'll update the PR now that the TEE tests have been added.  Shall I keep SEV / SNP out of the loop?

[the remark this answers is not in the document]
Comment (fitzthum) [reply src/runtime/Makefile:606]: I guess we can setthis  for SEV/SNP for now. We can update in the future if we get the memory hotplug sorted out.

[the remark this answers is not in the document]
Comment (fidencio) [reply src/runtime/Makefile:603]: @fitzthum, cool, I'm rebasing the PR atop of current `main` so we're able to run the tests on the TEE machines.
Mind to approve the PR, @fitzthum?

File: src/runtime/config/configuration-qemu-sev.toml.in (hunk 1 above)
Comment (fidencio) [reply src/runtime/config/configuration-qemu-sev.toml.in]: Yep, I didn't even notice this was part of the first PR (sorry), and I dropped changes related to this on this version.

If we're not using this anywhere, we should drop this part for sure.

**Comment by fidencio** [comment]
We need to backport this one to the CCv0 branch.

**Review by fitzthum — APPROVED** [review]

**Review by chavafg — APPROVED** [review]

**Review by amshinde — APPROVED** [review]

**Pull request merged**